packages/react-storage/src/components/StorageBrowser/Views/LocationsView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+
+import {
+  DataTable,
+  TABLE_DATA_BUTTON_CLASS,
+  TABLE_HEADER_BUTTON_CLASS_NAME,
+  TABLE_HEADER_CLASS_NAME,
+} from '../../../components/DataTable';
+import { useControl } from '../../../context/controls';
+import { useLocationsData } from '../../../context/actions';
+import { LocationAccess } from '../../../context/types';
+import { compareStrings } from '../../../context/controls/Table';
+import { ButtonElement, IconElement } from '../../../context/elements';
+import { capitalize } from '@aws-amplify/ui';
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState = {
+  selection: string;
+  direction: SortDirection;
+};
+
+const getCompareFn = (selection: string) => {
+  switch (selection) {
+    case 'scope':
+    case 'type':
+    case 'permission':
+      return compareStrings;
+  }
+};
+
+const getColumnItem = ({
+  key,
+  selection,
+  direction,
+  onTableHeaderClick,
+}: {
+  key: string;
+  selection: string;
+  direction: SortDirection;
+  onTableHeaderClick: (location: string) => void;
+}) => ({
+  children: (
+    <ButtonElement variant="sort" className={TABLE_HEADER_BUTTON_CLASS_NAME}>
+      {capitalize(key)}
+      <IconElement
+        variant={
+          selection === key && direction !== 'none'
```

Comment:
In what scenario is the `direction` check relevant if the `selection === key` is `true`?

Reply:
I'm not sure if the case could ever happen, but I added it because I was getting a typescript error that said `sort-none` wasn't a variant for `IconElement` and so this check helps with the typescript complaining